src/graph.py

```diff
@@ -0,0 +1,200 @@
+        while travel:
+            edge = travel.pop()[0]
+            if edge not in depth:
+                travel.extend(self.g[edge][::-1])
+                depth_list.append(edge)
+                depth.add(edge)
+        return depth_list
+
+    def breadth_first_traversal(self, start):
+        """Return a list of nodes based on breadth first traversal."""
+        start = (start, 0)
+        travel = [start]
+        depth = []
+        while travel:
+            edge = travel.pop(0)[0]
+            if edge not in depth:
+                travel.extend(self.g[edge])
+                depth.append(edge)
+        return depth
+
+    def warshall(self, start, finish):
+        """Floyd Warshall formula."""
+        if self._path(start, finish):
+            distance = {}
+            path = {}
+            graph = self.g
+            for node in graph:
+                distance[node] = {}
+                path[node] = {}
+                for visit in graph:
+                    distance[node][visit] = 1000
+                    path[node][visit] = -1
+                distance[node][node] = 0
+                for edge in range(len(graph[node])):
+                    key_edge = graph[node][edge][0]
+                    distance[node][key_edge] = graph[node][edge][1]
+                    path[node][key_edge] = node
+            for second_node in graph:
+                for node in graph:
+                    for visit in graph:
+                        new_distance = distance[node][second_node] + distance[second_node][visit]
+                        if new_distance < distance[node][visit]:
+                            distance[node][visit] = new_distance
+                            path[node][visit] = path[second_node][visit]
+            return self._return_path_warthal(path, start, finish)
+        raise KeyError('No such path.')
+
+    def dijkstra(self, start, finish):
+        """Dijkstra shortest path algorithm."""
+        if self._path(start, finish):
+            visited = {start: 0}
+            path = {}
+            nodes = set(self.breadth_first_traversal(start))
+            while nodes:
+                min_node = None
+                for node in nodes:
+                    if node in visited:
+                        if min_node is None:
+                            min_node = node
+                        elif visited[node] < visited[min_node]:
+                            min_node = node
+                    if min_node is None:
+                        break
+                nodes.remove(min_node)
+                curr_distance = visited[min_node]
+                for edge in self._edges_for_node(min_node):
+                    distance = curr_distance + edge[1][1]
+                    if edge[1][0] not in visited or distance < visited[edge[1][0]]:
+                        visited[edge[1][0]] = distance
+                        path[edge[1][0]] = min_node
+            return self._return_path(path, start, finish)
```

Comment:
You should return the distance as well in order to more completely test the function, (something like a tuple with distance, path).